### PR TITLE
fixes #3119

### LINKED ISSUE #3119: No longer able to publish a NETStandard.Library without a runtime directive
## Steps to reproduce
dotnet new
vi project.json
{
  "version": "1.0.0-*",
  "dependencies": {
    "NETStandard.Library" : "1.5.0-rc2-24027"
  },
  "frameworks": {
    "netstandard1.5": {
    }
  }
}
dotnet restore
dotnet build
dotnet publish

## Expected  behavior
Late RC2 builds would perform the publish and just the necessary dependencies would be published to the bin/<flavor>/netstandard1.5/publish directory

## Actual behavior
Can not find runtime target for framework '.NETStandard,Version=v1.5' compatible with one of the target runtimes: 'osx.10.11-x64'. Possible causes:
1. The project has not been restored or restore failed - run `dotnet restore`
2. The project does not list one of 'osx.10.11-x64' in the 'runtimes' section.

## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64

### PR DESCRIPTION
**Pull request by brthor**
cc @blackdwarf

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs [hunk 1]
```diff
@@ -66,7 +66,8 @@ public ProjectContext GetRuntimeContext(ProjectContext context, IEnumerable<stri
                 throw new InvalidOperationException($"Can not find runtime target for framework '{context.TargetFramework}' compatible with one of the target runtimes: '{rids}'. " +
                                                     "Possible causes:" + Environment.NewLine +
                                                     "1. The project has not been restored or restore failed - run `dotnet restore`" + Environment.NewLine +
-                                                    $"2. The project does not list one of '{rids}' in the 'runtimes' section.");
+                                                    $"2. The project does not list one of '{rids}' in the 'runtimes' section." + Environment.NewLine +
+                                                    "3. You may be trying to publish a library, which are not supported. Use `dotnet pack` to distribute libraries.");
```
Comment (joshfree) [inline src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs]: shouldn't that be "is" not supported

Comment (brthor) [inline src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs]: good catch

Comment (brthor) [inline src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs]: fixed, thanks!

**Comment by blackdwarf** [comment]
:shipit: 

Also, @jeffschwMSFT to take a look.

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by brthor** [comment]
@dotnet-bot test  Ubuntu16.04 x64 Debug Build  please
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
cc @mmitche 

> Disk has run out of space and cannot get latest changes. Please contact netciadmin@microsoft.com.

on Azure0602081142

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by mmitche** [comment]
@brthor Machine has a ton of space...hmmm

**Comment by brthor** [comment]
> 11:35:52 FATAL: Unable to produce a script file
> 11:35:52 java.io.IOException: Failed to create a temp file on /mnt/j/workspace/dotnet_cli/rel_1.0.0-preview2/release_ubuntu_x64_prtest

This seems to be the root issue, permissions problem?

**Comment by mmitche** [comment]
No, I just manually created it, and it was fine.

**Comment by mmitche** [comment]
@dotnet-bot test Ubuntu x64 Release Build

**Comment by mmitche** [comment]
Root disk has 100% usage

**Pull request merged**